gamification-evm-services/src/main/java/io/meeds/evm/gamification/scheduling/task/EvmContractSaveTask.java

```diff
@@ -55,25 +55,10 @@ public class EvmContractTransferTask {
   private EvmContractTransferService evmContractTransferService;
 
   @ContainerTransactional
-  @Scheduled(cron = "0 * * * * *")
-  public synchronized void listenEVMContractTransfer() {
-    try {
-      List<RuleDTO> filteredRules = evmContractTransferService.getFilteredEVMRules();
-      if (CollectionUtils.isNotEmpty(filteredRules)) {
-        filteredRules.forEach(rule -> {
-          evmContractTransferService.listenEvmContractTransfer(rule);
-        });
-      }
-    } catch (Exception e) {
-      LOG.error("An error occurred while rewarding for EVM events", e);
-    }
-  }
-
-  @ContainerTransactional
-  @Scheduled(cron = "0 * * * * *")
+  @Scheduled(cron = "${gamification.evm.transactionSave.cron:0 * * * * *}")
   public synchronized void saveEVMContractTransactions() {
     try {
-      List<RuleDTO> filteredRules = evmContractTransferService.getFilteredEVMRules();
+      List<RuleDTO> filteredRules = evmContractTransferService.getEnabledEvmRules();
```

Comment:
What if I disable a rule, then I enable it after few days, shouldn't we ensure that the Rule considers only newer transactions? We may need to add a listener on "rule enablement" to reset the rule/event start listening date. To check with PO please, what business logic to implement in this specific use case.